qctoolkit/pulses/pulse_template.py

```diff
@@ -190,19 +197,25 @@ def build_sequence(self,
             instruction_block.add_instruction_meas(measurements)
             instruction_block.add_instruction_exec(waveform)
 
-    def _internal_create_program(self,
+    def _internal_create_program(self, *,
                                  parameters: Dict[str, Parameter],
                                  measurement_mapping: Dict[str, Optional[str]],
                                  channel_mapping: Dict[ChannelID, Optional[ChannelID]],
                                  parent_loop: Loop) -> None:
         # todo (2018-07-05): why are parameter constraints not validated here?
-        parameters = {parameter_name: parameter_value.get_value()
-                      for parameter_name, parameter_value in parameters.items()
-                      if parameter_name in self.parameter_names}
+        try:
+            parameters = {parameter_name: parameters[parameter_name].get_value()
+                          for parameter_name in self.parameter_names}
+
+            measurement_parameters = {parameter_name: parameters[parameter_name].get_value()
+                                      for parameter_name in self.measurement_parameters}
+        except KeyError as e:
+            raise ParameterNotProvidedException(str(e)) from e
```

Comment:
Why do you create an exta dictionary for the measurement parameters? They should be a subset of `parameter_names`

Reply:
should be but currently aren't according to https://github.com/qutech/qc-toolkit/issues/230